### PR TITLE
bugfix/completions-stop-when-process-exits-while-waiting-for-a-response

### PR DESCRIPTION
**Pull request by amircodota**
… to the response

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -271,31 +271,71 @@ class TabNine {
...
         }
-        this.rl.once('line', (response) => {
+        const onResponse: (input: any) => void = (response) => {
           let any_response: any = JSON.parse(response.toString());
           resolve(any_response);
-        });
+        };
+        this.rl.once('line', onResponse);
+
+        unregisterFunctions.push(() => this.rl.removeListener('line', onResponse));
```
Comment (dimacodota) [inline src/extension.ts:292]: do we need to unregister events that were register with `once` ?

Comment (amircodota) [reply src/extension.ts:292]: Yes, if the promise was resolved or rejected, and these were not fired yet, we need to unregister or we are going to leak.
Specifically, the process does not exit under normal circumstances, so if we just register for it with evey request, we are going to leak

Comment (dimacodota) [reply src/extension.ts:292]: 👍

Comment (dimacodota) [reply src/extension.ts:292]: nice catch...

**Pull request merged**